src/concrete_convex_hull.jl

```diff
@@ -136,6 +119,48 @@ function convex_hull!(points::Vector{VN};
     end
 end
 
+function _two_points_2d!(points)
+
+    n = length(first(points))
+
+    if n == 1
+        p1, p2 = points[1], points[2]
+        if p1 == p2  # check for redundancy
+            pop!(points)
+        elseif p1[1] > p2[1]
+            points[1], points[2] = p2, p1
+        end
+    elseif n == 2
+        # special case, see #876
+        p1, p2 = points[1], points[2]
+        if p1 == p2  # check for redundancy
+            pop!(points)
+        elseif p1 <= p2
+            nothing
+        else
+            points[1], points[2] = p2, p1
+        end
+    end
+    return points
+end
+
+function _three_points_2d!(points)
+    A, B, C = points[1], points[2], points[3]
+    turn = (A[2] - B[2]) * C[1] + (B[1] - A[1]) * C[2] + (A[1] * B[2] - B[1] * A[2])
+
+    if isapproxzero(turn)
+        # ABC are collinear
+        points[1], points[2] = points[argmin([A[1], B[1], C[1]])], points[argmax([A[1], B[1], C[1]])]
```

Comment:
I think the following is equivalent.
```suggestion
        points[1], points[2] = min([A[1], B[1], C[1]]), max([A[1], B[1], C[1]])
```
However, I do not understand why it is sufficient to sort by x1 coordinate. What if they all lie on the line `x1 = c` for some `c`?

Reply:
Sorry, I was wrong here. Can you undo again? :wink:

Reply:
I am also still concerned that this is incorrect if the x1 value is identical (then you probably get the same point twice).